fdmi/plugins/motr-fdmi-app.md

```diff
@@ -1,33 +1,70 @@
 
-In this folder, we have an example of an FDMI application (fdmi_sample_plugin) that will be compiled as part of the initial Motr compilation using the Makefile.am dependency and the Makefile.sub file.
+This extension interface is really powerful since it gives more flexibility to developers so they can easily modify and deploy in CORTX independently on additional and dedicaded nodes without compromising the fast path for Motr. 
 
-To run a simple test of this example, you can execute the following command inside this directory:
+In addition, this interface inherits the horizontal scalability property form motr while being reliable since it will be transactionally coupled with the core.
+  
+FDMI applications can be seen as traditional object or key-value applications that has a more closed relationship with CORTX-Motr. 
 
-`./fdmi_plugin_st.sh`
+# FDMI Implementations Overview
+
+FDMI is a scalable publish-subscribe interface where each Motr instance works as a source and produces records that describe operations. 
+
+On the other side, there is a FDMI Application or Plugin that registers a filter in the Motr Filter Database to selects records that match such filters.
+
+The intial substrings to match metadata fields are setup at the CORTX initialization time.
+
+The FDMI source instances sends matching records to the FDMI Application or Plugin in batches with their transactional contexts.
+
+For each record, the FDMI Application or Plugin performs actions and sends acknowledgements back to the source instances using Motr RPC calls.
+
+# How to create and FDMI Application and run with examples
+
+All FDMI related service code and examples are in the `cortx-motr/fdmi` directory. 
+
+One of the FDMI Application example is within the `plugins` directory named as fdmi_sample_plugin. 
 
-This shell script starts Motr services using a specific filter configuration, starts the fdmi_plugin_sample application to listen for specific records that match with the filter already setup and do some key-value operations that reflects this specifications. 
+This application was written in C using `motr/client.h` interface and it adds the new Motr client to the cluster and start listening for FDMI records or events that matches the FDMI filter substrings.  
 
-The output will show whether the test was successful or not. 
+The executable will be compiled as part of the initial Motr compilation using the Makefile.am dependency and the Makefile.sub file.
 
-A second way to test this fdmi_sample_plugin will be by directly running the fdmi_sample app wrapper with the following command:
+This program then communicates with the fdmi_app python wrapper by printing to stdout the records. 
+
+To test this application we need to run the fdmi_sample app wrapper typing in the console the following command:
```

Comment:
To test this application we need to run the fdmi_app wrapper in the terminal with the following command: